pkg/instancetype/find/cluster_instancetype.go

```diff
@@ -45,18 +45,30 @@ func (f *clusterInstancetypeFinder) Find(vm *virtv1.VirtualMachine) (*v1beta1.Vi
 	if vm.Spec.Instancetype == nil {
 		return nil, nil
 	}
+	return f.findByName(vm.Spec.Instancetype.Name)
+}
+
+func (f *clusterInstancetypeFinder) FindFromVMI(vmi *virtv1.VirtualMachineInstance) (*v1beta1.VirtualMachineClusterInstancetype, error) {
+	instancetypeName, ok := vmi.GetLabels()[virtv1.ClusterInstancetypeAnnotation]
+	if !ok {
+		return nil, fmt.Errorf("unable to find instance type annotation on VMI")
```

Comment:
Is it correct to return an error? A VM without instancetype will return `nil, nil`, as seen above.